tests/unittests/tests-uri_parser/tests-uri_parser.c

```diff
@@ -21,599 +21,601 @@
+        .scheme = "a1b2",
+    };
+
+    expected_uri_result_t *test_vec[] = {
+        &very_long_scheme,
+        &normal_scheme,
+        &short_scheme,
+        &digit_scheme,
+    };
+
+    uri_parser_result_t uri_res;
+    int ret = 0;
+
+    for (unsigned int i = 0; i < ARRAY_SIZE(test_vec); ++i) {
+        ret = uri_parser_process_string(&uri_res, test_vec[i]->input_uri);
+        if (ret != expected_ret) {
+            /* if the uri_parser return indicates an error, this test fails */
+            _print_return_expectation(test_vec[i]->input_uri, expected_ret, ret);
+            TEST_FAIL(failure_msg);
+        }
+        else {
+            /* if the uri_parser return indicates success
+             * the length of the schemes must match.
+             * we can't use strlen on the result scheme as it is not null terminated,
+             * but the uri_parser provides the length separately*/
+            if (uri_res.scheme_len != strlen(test_vec[i]->scheme)) {
+                printf(
+                    "With given input uri '%s', expected a scheme with the length '%d' but got '%d'\n",
+                    test_vec[i]->input_uri, strlen(test_vec[i]->scheme), uri_res.scheme_len);
+                TEST_FAIL(failure_msg);
+            }
+            else {
+                /* If the schemes have the same length, they also should look identical */
+                if (strncmp(uri_res.scheme, test_vec[i]->scheme, uri_res.scheme_len) != 0) {
+                    printf("With given input uri '%s', expected scheme '%s' but got '%.*s'\n",
+                           test_vec[i]->input_uri, test_vec[i]->scheme, uri_res.scheme_len,
+                           uri_res.scheme);
+                    TEST_FAIL(failure_msg);
+                }
+            }
+        }
+    }
+}
+
+/* ABNF of scheme:
+ * ALPHA *( ALPHA / DIGIT / "+" / "-" / "." ) ":"
+ */
+static void _error_if_scheme_is_invalid(void)
+{
+    char *failure_msg = "Failure: The uri_parser failed to reject an invalid scheme.";
+    int expected_ret = -1;  /* Indicates an invalid uri */
+
+    char *space_in_scheme = "co ap://example.org/";
+    char *underscore_in_scheme = "co_ao://example.org/";
+    char *slash_in_scheme = "co/ap://example.org";
+    char *binary_garbage_in_scheme = "co\x01\x08ap://example.org/";
+    char *not_starting_with_letter = "91coap://example.org";
+
+    char *test_vec[] = {
+        space_in_scheme,
+        underscore_in_scheme,
+        slash_in_scheme,
+        binary_garbage_in_scheme,
+        not_starting_with_letter,
+    };
+
+    uri_parser_result_t uri_res;
+    int ret = 0;
+
+    for (unsigned int i = 0; i < ARRAY_SIZE(test_vec); ++i) {
+        ret = uri_parser_process_string(&uri_res, test_vec[i]);
+        if (ret != expected_ret) {
+            /* At the moment, the uri_parser is not capable of detecting invalid schemes
+             *_print_return_expectation(test_vec[i], expected_ret, ret);
```

Comment:
Do we have some kind of TODO tag so this doesn't get lost?

Reply:
Removed from the PR. I will add this test later as soon as the uri-parser gets an appropriate update :)